pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -104,6 +104,29 @@ public Consumer<T> subscribe() throws PulsarClientException {
         }
     }
 
+    private CompletableFuture<Boolean> topicExists(String topic) {
+        CompletableFuture<Boolean> existsFuture = new CompletableFuture<>();
+        client.getPartitionedTopicMetadata(topic, false).thenAccept(metadata -> {
+            TopicName topicName = TopicName.get(topic);
+            if (topicName.isPersistent()) {
+                existsFuture.complete(true);
```

Comment:
If the topic is persistent, so don't need to get the metadata ?

Reply:
This method is used to check the topic created for the old `DLQ` rule(`{namespace}-{subscription}-DLQ`) already exists
- if yes, use the old topic
- else, create DLQ with the new rule(`{topic}-{subscription}-DLQ`)

So if it is a persistent topic, the result is `true` whether it is partitioned or not. If it is a non-persistent topic, the result is `false` if it is a non-partitioned topic(the non-partitioned DLQ will be removed automatically after the consumers are offline)

Reply:
I renamed the method to make it more clear

Reply:
ok